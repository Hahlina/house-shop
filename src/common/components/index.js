export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
