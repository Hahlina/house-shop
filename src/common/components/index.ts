export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
export * from './Container/Container';
export * from './Image/Image';
export * from './ErrorPage/ErrorPage';
export * from './SomethingWentWrong/SomethingWentWrong';
export * from './NothingFound/NothingFound';
