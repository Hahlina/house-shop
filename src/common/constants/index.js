export * from './config';
export * from './apiRoutes';
export * from './initValues';
export * from './appRoutes';
