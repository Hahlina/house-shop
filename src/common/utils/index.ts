export * from './convertPrice';
export * from './capitalizeFirstLetter';
