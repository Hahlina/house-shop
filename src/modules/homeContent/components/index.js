export * from "./PreviewBanner/PreviewBanner";
export * from "./PropertiesList/PropertiesList";
