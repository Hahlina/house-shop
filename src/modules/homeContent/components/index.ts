export * from './PreviewBanner/PreviewBanner';
export * from './PropertiesList/PropertiesList';
export * from './PropertiesPagination/PropertiesPagination';
