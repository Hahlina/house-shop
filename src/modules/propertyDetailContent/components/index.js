export * from "./PropertyGallery/PropertyGallery";
export * from "./FullInformationCard/FullInformationCard";
