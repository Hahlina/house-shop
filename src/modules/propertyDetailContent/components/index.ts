export * from './PropertyGallery/PropertyGallery';
export * from './PropertyGallery/PropertyGalleryLoader';
export * from './FullInformationCard/FullInformationCard';
export * from './FullInformationCard/FullInformationCardLoader';
export * from './PropertyDetailContentLoader/PropertyDetailContentLoader';
export * from './PropertyDetailBreadCrumbs/PropertyDetailBreadCrumbs';
