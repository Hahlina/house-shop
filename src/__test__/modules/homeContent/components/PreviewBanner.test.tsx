import { PreviewBanner } from 'modules/homeContent/components';

import { render, screen } from 'common/utils/test';
import { basePlaceholderBase64 } from 'common/constants';

describe('PreviewBanner components', () => {
    it('should render a preview banner with an placeholders', () => {
        render(<PreviewBanner />);
        const image = screen.getByAltText('home-banner');
        expect(image).toHaveAttribute('src', basePlaceholderBase64);
    });

    it('should render a preview banner title', () => {
        render(<PreviewBanner />);
        const title = screen.getByText(/Let's Find Your/i);
        expect(title).toBeInTheDocument();
    });

    it('should have height - 100svh', () => {
        const { container } = render(<PreviewBanner />);
        expect(container).toHaveStyle('height: 100svh');
    });

    it('should take a snapshot', () => {
        const { container } = render(<PreviewBanner />);
        expect(container).toMatchSnapshot();
    });
});
