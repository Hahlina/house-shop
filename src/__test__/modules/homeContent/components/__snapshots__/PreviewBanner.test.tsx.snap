// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewBanner components should generate a math snapshot 1`] = `
<div>
  <div
    class="sc-beySPh cjvRJa"
  >
    <img
      alt="house"
      class="preview-image"
      src="/images/previewImage.jpg"
    />
    <h1
      class="MuiTypography-root MuiTypography-body1 preview-title css-ahj2mt-MuiTypography-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Let's Find Your
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Dream House
      </p>
    </h1>
  </div>
</div>
`;
